include/oneapi/dpl/pstl/algorithm_ranges_impl.h

```diff
@@ -1160,13 +1163,15 @@ __pattern_mismatch(_Tag __tag, _ExecutionPolicy&& __exec, _R1&& __r1, _R2&& __r2
     auto __first1 = std::ranges::begin(__r1);
     auto __first2 = std::ranges::begin(__r2);
 
-    return oneapi::dpl::__internal::__pattern_mismatch(__tag, std::forward<_ExecutionPolicy>(__exec), __first1,
-                                                       __first1 + std::ranges::size(__r1), __first2,
-                                                       __first2 + std::ranges::size(__r2), __bin_pred);
+    const auto& [first, second] = oneapi::dpl::__internal::__pattern_mismatch(
+        __tag, std::forward<_ExecutionPolicy>(__exec), __first1, __first1 + std::ranges::size(__r1), __first2,
+        __first2 + std::ranges::size(__r2), __bin_pred);
```

Comment:
Why is this change necessary? Looks like `__pattern_mismatch` returns a `__mismatch_return_t` as well.